qutip/core/qobj.py

```diff
@@ -1545,8 +1545,9 @@ def eigenstates(
         eigvals: int = 0,
         tol: float = 0,
         maxiter: int = 100000,
-        phase_fix: int = None
-    ) -> tuple[np.ndarray, list[Qobj]]:
+        phase_fix: int = None,
+        output_type: Literal['kets', 'oper'] = 'kets'
+    ) -> tuple[np.ndarray, Union[np.ndarray[Qobj], Qobj]]:
```

Comment:
This is not very useful type hints...
When multiple output types are possible, using `@overload` is better in this case.